app/models/transaction.rb

```diff
@@ -4,4 +4,14 @@ class Transaction < ApplicationRecord
   belongs_to :invoice
 
   enum result: %w(success failed)
-end
\ No newline at end of file
+
+  scope :successful, -> { where(result: 'success') }
+
+  def self.on_date(date)
```

Comment:
Since we're adding a scope and a class method to Transaction, could we include some tests (they can be fairly rudimentary) for those functions as well.